### PR TITLE
fix: resolve failing CI

### PR DESCRIPTION
**Pull request by Borda**
fixing several issues:
- leaked installed minimal dependencies
- common pytest configurations
- add testing aggregated check
- fix for RinIf Thunder
- fix guarded imports in extensions

### CODE REVIEW

File: extensions/thunder_gpt/README.md [hunk 1]
```diff
@@ -444,7 +444,7 @@ We provide ready-to-use Fabric strategies that integrate Thunder DDP|FSDP. Under
 ```python
 model = thunder.distributed.ddp(model)
 # or
-# model = thunder.distributed.fsdp(model)
+# model = thunder_gpt.distributed.fsdp(model)
```
Comment (Borda) [inline extensions/thunder_gpt/README.md:447]:
```suggestion
# model = thunder.distributed.fsdp(model)
```

File: extensions/thunder_gpt/pretrain.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def setup(
     tokenizer_dir: Optional[Path] = None,
     logger_name: Literal["wandb", "tensorboard", "csv"] = "tensorboard",
     seed: int = 42,
-    compiler: Optional[Literal["thunder", "torch"]] = "thunder",
+    compiler: Optional[Literal["thunder_gpt", "torch"]] = "thunder_gpt",
```
Comment (Borda) [inline extensions/thunder_gpt/pretrain.py:72]:
```suggestion
    compiler: Optional[Literal["thunder", "torch"]] = "thunder",
```

File: extensions/thunder_gpt/strategies/thunder_ddp.py [hunk 3]
```diff
@@ -22,18 +22,17 @@
...
 )
 from lightning.fabric.utilities.rank_zero import rank_zero_only
-from lightning_utilities.core.imports import RequirementCache
 from lightning_utilities.core.rank_zero import rank_zero_only as utils_rank_zero_only
 from torch import Tensor
 from torch.nn import Module
 from typing_extensions import override
 
+from litgpt.utils import _THUNDER_AVAILABLE
+
```
Comment (Borda) [inline extensions/thunder_gpt/strategies/thunder_ddp.py:31]:
```suggestion
```

File: tests/conftest.py [hunk 4]
```diff
@@ -103,20 +103,17 @@ def longform_path(tmp_path):
     path = tmp_path / "longform"
     path.mkdir()
     for split in ("train", "val"):
-        file = Path(__file__).parent / "data" / "fixtures" / f"longform_{split}.json"
+        file = Path(__file__).parent / "data" / ("_fixtures") / f"longform_{split}.json"
```
Comment (Borda) [inline tests/conftest.py:106]:
```suggestion
        file = Path(__file__).parent / "data" / "_fixtures" / f"longform_{split}.json"
```

**Comment by Borda** [comment]
@t-vi could you pls update the required checks to be new Azure and `testing-guardian`

**Review by robTheBuildr — APPROVED** [review]

File: extensions/thunder_gpt/README.md [hunk 5]
```diff
@@ -460,7 +460,7 @@ After applying the DDP transformation, the backward trace will include the expec
 With `L.Fabric`, this is how to use them:
 
 ```python
-from extensions.thunder.strategies import ThunderFSDPStrategy, ThunderDDPStrategy
+from extensions.thunder_gpt.strategies import ThunderFSDPStrategy, ThunderDDPStrategy
```
[the remark this answers is not in the document]
Comment (t-vi) [reply extensions/thunder_gpt/README.md:463]: OK, so the reason is that "extensions" is bad in the global namespace? then maybe litgpt.extensions would be better?
